shopfloor/services/zone_picking.py

```diff
@@ -465,11 +475,15 @@ def list_move_lines(self):
         """
         return self._list_move_lines(self.zone_location)
 
-    def _list_move_lines(self, location, product=False):
+    def _list_move_lines(self, location, product=False, sublocation=False):
         move_lines = self._find_location_move_lines(location, product=product)
-        return self._response_for_select_line(move_lines, product=product)
+        return self._response_for_select_line(
+            move_lines, product=product, sublocation=sublocation
+        )
 
-    def _scan_source_location(self, barcode, confirmation=False, product_id=False):
+    def _scan_source_location(
+        self, barcode, confirmation=False, product_id=False, sublocation=False
```

Comment:
This is because the `scan_source` implementation iterates over a list of handlers https://github.com/OCA/wms/blob/14.0/shopfloor/services/zone_picking.py#L657